Adjusts the testing strategy document to adhere to typical quality standards (#242)
kyma-project/telemetry-manager: docs/contributor/testing.md

​
| Phase | When | How | Result |
| --- | --- | --- | --- |
| Requirement analysis | Sprint planning, grooming and triaging meetings | Brainstorm the feature implementation, design and its implications on the testability. | A clarified implementation design with a focus on testability, the acceptance tests criteria are collected, and the testing surface is defined as a part of the story. |

> | Test planning, design and development | Development phase during the sprint | The new functionality is covered with sufficient Unit, Integration and Acceptance-tests. | The unit and integration test suites are augmented, and new tests are integrated into the CI pipeline. All new functionality is covered with an Acceptance test. |

Reviewer: ```suggestion
| Test planning, design and development | Development phase during the sprint | The new functionality is covered with sufficient Unit, Integration, and Acceptance tests. | The unit and integration test suites are augmented, and new tests are integrated into the CI pipeline. All new functionality is covered with an Acceptance test. |
```

> | Test validation | Acceptance phase during the sprint | The correctness and completeness of the Acceptance Criteria coverage by the Test Suite are verified during the code review and iteration review. | The Acceptance Criteria get integrated into the E2E test suite and become a regression suite for future verification. |

Reviewer: ```suggestion
| Test validation | Acceptance phase during the sprint | During the code review and iteration review, verify that the Test Suite correctly and completely covers the Acceptance Criteria. | The Acceptance Criteria are integrated into the E2E test suite and become a regression suite for future verification. |
```

Clearer phrasing with active voice, fewer nouns, more verbs.

| Unit (located with the individual source files) | Unit | It tests the units in isolation. This test suite assesses the implementation correctness of the units of business logic. |
| Env-tests (located with the individual source files) | Integration  (low-level) | It tests the behaviour of the Telemetry Manager in integration with a Kubernetes API server replaced with a test double. This test suite assesses the integration correctness of the Telemetry Manager. |
| [E2E](/test/e2e) | Acceptance / Integration (high-level) | It tests the usability scenarios of the Telemetry Manager in a cluster. This test suite assesses the functional correctness of the Telemetry Manager. |
| Operational | System | It validates the operational aspects of the module (successful module upgrades, etc.). |
​
#### Testing of new functionality
​

> Testing a new functionality encompasses two activities: capturing the acceptance criteria for each new functionality and providing the requirements traceability. The `Requirement` ↔︎ `Test Suite` linkage is achieved by the Behavioural-Driven Testing DSL of the [Ginkgo](https://onsi.github.io/ginkgo/) testing framework. The `User Story` ↔︎ `Acceptance Criteria` linkage is provided semi-automatic by making acceptance criteria a mandatory part of each story with new functionality (using the PR-template checklist).

Reviewer: ```suggestion
Testing a new functionality encompasses two activities: capturing the acceptance criteria for each new functionality and providing the requirements traceability. `Requirement` and `Test Suite` are linked by the Behavioural-Driven Testing DSL of the [Ginkgo](https://onsi.github.io/ginkgo/) testing framework. `User Story` and `Acceptance Criteria` are linked semi-automatically because acceptance criteria are a mandatory part of each user story with new functionality (using the PR-template checklist).
```

Why do you love nouns so much?? :D

Author: I reckon that is the way I experience the world around me :) I blame object-oriented analysis and domain-driven design for this.

​
### Source Code Quality
​

> Most of the source code quality prerequisites are checked in an automated way by following:

Reviewer: ```suggestion
Most of the source code quality prerequisites are checked automatically in the following ways:
```
